README.md

```diff
@@ -119,6 +119,62 @@ ClientConfig = [{reconnect_cool_down_seconds, 10}],
 ok = brod:start_client([{"localhost", 9092}], brod_client_1, ClientConfig).
 ```
 
+## Handling of connection errors
+
+Usually you would have a supervisor that starts your worker module. In the worker's `start_link` method, you should do two things:
+
+1. Make sure the Brod client you're going to use is running by calling `brod:start_client`. That will start the client under Brod's supervisor, if a client with the given name does not already exist. Afterwards, Brod will automatically recover from connection issues.
+1. Start, e.g., a group subscriber by calling `brod:start_link_group_subscriber`, and return the pid for your supervisor to monitor.
+
+(Note that using `brod:start_link_client` to link the client to your supervisor is deprecated. Using it, a connection loss won't trigger a reconnect, even though the process linked to stays alive.)
+
+Also, the subscriber will fail very quickly in case the brod client can't connect, which causes the supervisor to give up and potentially bring down the VM. In order to mitigate this, an exponential back-off strategy is needed, or at least a delay, between calls to `brod:start_link_group_subscriber`. In Elixir this may look like this:
+
+```Elixir
+defmodule MyGroupSubscriber do
+  @doc """
+  Makes sure the Brod client is running and starts the group subscriber.
+  """
+  def start_link do
+    brod_client_id = :my_brod_client
+    :ok = :brod.start_client([localhost: 9092], brod_client_id)
+    safely_start_group_subscriber(brod_client_id, "my-consumer-group", ["some-topic"])
+  end
+
+  # Starts the group subscriber, using exponential back-off if the brod client is down.
+  defp safely_start_group_subscriber(
+      brod_client_id, consumer_group_id, topics,
+      restart_delay_ms \\ 100, max_restart_delay_ms \\ 10_000
+  ) do
+    result = start_group_subscriber(brod_client_id, consumer_group_id, topics)
```

Comment:
It would be easier to simply match `{:ok, _}` here and 'let it crash' if kafka cluster is down.
Delayed retry can be implemented in supervisor, here is an example:
https://github.com/klarna/kastlex/blob/master/lib/kastlex/collectors_sup.ex#L25

Or you can implement a genserver to monitor group subscriber and restart it after
some exponential delay (i.e. not only handle the start_link errors),  but also handle
crashes after it is successfully started.